kernel/src/mem.rs

```diff
@@ -50,6 +50,19 @@ pub trait Read {
     /// address `0x0`.
     fn ptr(&self) -> *const u8;
 
+    /// Applies a function to the (read only) slice reference pointed
+    /// to by the AppSlice.
+    ///
+    /// If the process is no longer alive and the memory has been
+    /// reclaimed, this method must return the default value.
+    ///
+    /// # Default AppSlice
+    ///
+    /// A default instance of an AppSlice must not execute the closure.
+    fn map<F>(&self, fun: F)
+    where
+        F: FnOnce(&[u8]);
+
```

Comment:
All the added functions in this file are no longer used, but still present here

Reply:
Done.